src/waveform/renderers/waveformwidgetrenderer.cpp

```diff
@@ -305,3 +309,25 @@ WaveformMarkPointer WaveformWidgetRenderer::getCueMarkAtPoint(QPoint point) cons
     }
     return nullptr;
 }
+
+// TODO(hacksdump): Optimize this search
+Beat WaveformWidgetRenderer::getBeatAtPoint(QPoint point) const {
+    for (const auto& pBeat : m_beatPositions.keys()) {
+        float position = m_beatPositions[pBeat];
+        if (getOrientation() == Qt::Horizontal) {
+            if (position - 5 < point.x() && point.x() < position + 5) {
+                return pBeat;
+            }
+        } else {
+            if (position - 5 < point.y() && point.y() < position + 5) {
+                return pBeat;
+            }
+        }
+    }
+    // Maybe consider using std::optional return type for this function.
+    // Optional wrapper should be suitable for searches.
+    Beat beat;
+    // Currently relying on setting enabled status to false.
+    beat.setEnabled(false);
```

Comment:
I was confused what this "enabled" state is for in the new Beat class. Is this all it is used for? If so, please get rid of it and use std::optional here.

Reply:
The `enabled` state is not introduced by the new Beat class, it was existing in the Beat protobuf. Considering this check, I will wrap the return type with std::optional.

Reply:
I have no idea what it means for a beat to be "enabled"... was this previously used in the code anywhere? If not, let's remove it.